src/cpp/src/partition_manager.cpp

```diff
@@ -249,12 +303,16 @@ shared_ptr<ModifyTimingInfo> PartitionManager::add(
                       << " into partition " << pid << std::endl;
         }
 
+
+        std::shared_ptr<arrow::Table> filtered_table_result = filterRowById(attributes_table, id_accessor[i]);
         partition_store_->add_entries(
             pid,
             /*n_entry=*/1,
             id_ptr + i,
-            code_ptr + i * code_size_bytes
+            code_ptr + i * code_size_bytes,
+            filtered_table_result
```

Comment:
This might be a significant performance slowdown since we do this filtering per entry.